### PR TITLE
Filters for Store List

### PR DESCRIPTION
**Pull request by anniero98**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

# What's new in this PR

### In sum
- Added filters for the store list: "Open now", "Products in stock", "EBT", "WIC", "Snap Match", and "Healthy Rewards Accepted".
- Partially waiting on finalized designs, but in its current state it is usable and conforms more-or-less

### Functionality
- Added state object `filters` with various property names that map directly to `store` object properties (except for the `openNow` property).
- Added horizontal scrollview of `Chip` components, selectable
- Implemented filtering so that filters stack, and search is also applied
- Realized that, in a similar manner to how we initially implemented Auth button permissions, we don't need to keep filtered stores in state. Rather, we can just calculate and update the list in the `render` function, since all the information we need is in state.

### Styling
- Updated the chips styling in `styled/store.js`. Most of the work was really done in #98, I just messed around with the heights and margins until things seemed alright.

- Instead of using the `Chip` `built-in` `selected` property, I implemented filter-selection with booleans, since we needed them in state to filter the stores correctly anyway. Also, I couldn't see a good way to switch on `filters[name]` for the styling, text color, _and_ icon color.

### Other details
- Bug fix: moved the formatting for `formatPhoneNumber` into `mapUtils/updateStoreData` into creation of a `phoneNumber` property for stores (when initializing the data)

- In `StoreDetailsScreen`, had to un-format it https://stackoverflow.com/questions/1862130/strip-all-non-numeric-characters-from-string-in-javascript

- Cleanup: created a `storeOpenStatus` property for stores in `mapUtils/updateStoreData` and called `computeOpenHours` to initialize it. This is a string; if it contains `Open` then store is currently open. Removed logic explicitly calling this function outside of `mapUtils`

- Added `productsInStock`, `snapOrEbtMatching`, `couponProgramPartner`, etc boolean fields.
**Airtable does not return fields which are empty / null from the API, so we need to initialize these ourselves if we don't want to run into errors where we expect record objects to contain certain properties**

[//]: # "Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

### Online sources

Referenced this: https://medium.com/better-programming/creating-a-multi-filter-function-to-filter-out-multiple-attributes-javascript-react-rails-5aad8e272142 (Has a paywall, though)

[//]: # "Optional - copy links to any tutorial or documentation that was useful to you when working on this PR"

### Related PRs

[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

## How to review

Sanity-check that the stores that show up in the list are as expected. Luckily, #98 makes testing this really easy (': Search strings should also stack, as shown in screen recording. 

[//]: # "The order in which to review files and what to expect when testing locally"

## Next steps

Styling: Designs aren't finalized, and though I more or less followed Figma for the initial pixel values the filter chips still seem kind of small.

Code: Pretty sure I should extract the filter chips to a component because the code is so repetitive it drives me a little crazy. At this moment in time I am not really sure how to though since they all have their own functions and icons etc... bit of a headache. No matter what we'll have to do `filters.<filter-type-name>` everywhere. But if we're in a rush to merge this we can refactor it later.

[//]: # "What's NOT in this PR, doesn't work yet, and/or still needs to be done"

## Tests Performed, Edge Cases

[//]: # "Hopefully we will add a testing suite/CI soon, but until then note down the steps you took to test locally"

### Screenshots

image

https://www.loom.com/share/7e47a6ca711b4b139da5b6af4cc07097

[//]: # "Add screenshots of expected behavior - GIFs if you're feeling fancy!"

CC: @wangannie

[//]: # "This tags in both Annies as a default. Feel free to change, or add on anyone who you should be in on the conversation."

### CODE REVIEW

File: screens/map/StoreListScreen.js [hunk 1]
```diff
@@ -94,8 +111,173 @@ export default class StoreListScreen extends React.Component {
             />
           </ColumnContainer>
         </NavHeaderContainer>
+        <ScrollView horizontal style={{ height: 52 }}>
```
Comment (wangannie) [inline screens/map/StoreListScreen.js:114]: Hide the scrollbar

File: styled/store.js [hunk 2]
```diff
@@ -107,23 +107,42 @@ export const styles = StyleSheet.create({
   input: {
     fontFamily: 'poppins-regular',
   },
-  chip: {
+  tagChip: {
     backgroundColor: Colors.lightestGreen,
     color: Colors.darkerGreen,
     marginRight: 6,
-    marginBottom: 4,
```
Comment (wangannie) [inline styled/store.js:114]: Wait we need this! That's A & S looked super weird with 2 lines of tags

Comment (anniero98) [reply styled/store.js:114]: wait I only removed it because it seems very odd to have both `marginVertical` and `marginBottom` and I expected that `marginVertical` overrode the latter - why wasn't it just `marginBottom` and `marginTop`?

Comment (wangannie) [reply styled/store.js:114]: I had to set `marginVertical` to 0 to override the defaults https://github.com/callstack/react-native-paper/blob/master/src/components/Chip.tsx#L333-L338 because it wasn't shrinking to my custom height (see my message in #ask), and I needed it to have 0 top margin.
image


There's probably a cleaner way to do this but I'm not super sure what combination of these margins work best.

Comment (anniero98) [reply styled/store.js:114]: ok re-ordered them for my sanity but if it works it works!

**Comment by anniero98** [comment]
> also yes I will work on componentalizing the chips
@thumn I will try to get some advice!! I bet someone knows how to but anyway dw about it for now!
